src/BinDeps.jl

```diff
@@ -3,15 +3,13 @@ __precompile__()
 module BinDeps
 
 using Compat
-using Compat: String, KERNEL
 
 export @make_run, @build_steps, find_library, download_cmd, unpack_cmd,
-        Choice, Choices, CCompile, FileDownloader, FileRule,
-        ChangeDirectory, FileDownloader, FileUnpacker, prepare_src,
-        autotools_install, CreateDirectory, MakeTargets, SystemLibInstall
+    Choice, Choices, CCompile, FileDownloader, FileRule,
+    ChangeDirectory, FileDownloader, FileUnpacker, prepare_src,
+    autotools_install, CreateDirectory, MakeTargets, SystemLibInstall
 
-const dlext = Libdl.dlext
-const shlib_ext = dlext # compatibility with older packages (e.g. ZMQ)
```

Comment:
we should really update all the packages that still depend on this 3 year old definition